src/App.js

```diff
@@ -1,25 +1,78 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import contacts from './contacts.json';
 
 function App() {
+  //Iteration 1: State with only the first 5
+ const fiveContacts = contacts.slice(0,5);
+ const [contactsState, setContacts] = useState (fiveContacts);
+
+//Iteration 2
+function addRandomContacts(){
```

Comment:
Your solution was adding an additional empty row, I revised it so it wouldn't anymore.
```
function addRandomContacts(){
  const randomNumber = Math.floor(Math.random() * contacts.length);
  const contactStateCopy = [...contactsState];
  contactStateCopy.push(contacts[randomNumber]);
  setContacts(contactStateCopy);
}